### PR TITLE
Django 1.7 compatibility

### PR DESCRIPTION
**Pull request by thoas**
- Export signals handling to models.py
- Export all settings to one unique file
- Export keyfmt to utils.py
- We are not importing models anymore in **init**.py which can cause in several cases multiple fails from django.apps Registry
- Refactor unittests using django own DiscoverRunner

Currently all tests pass except SwitchTests.test_no_query which I'm missing something why we should make on query as we don't cache query from Django itself (we only cache proxies from `waffle.__init__.py`).

I'm available if you want more information.

This PR is currently under development and not available to merge btw.

### CODE REVIEW

File: test_settings.py [hunk 1]
```diff
@@ -7,7 +8,10 @@
 DEBUG = True
 TEMPLATE_DEBUG = True
 
-TEST_RUNNER = 'django_nose.runner.NoseTestSuiteRunner'
+if django.VERSION <= (1, 6):
```
Comment (ghickman) [inline test_settings.py]: @thoas – I think this should be `<`? django.test.runner.DiscoverRunner was added in 1.6.

**Comment by ashchristopher** [comment]
@thoas, @jsocol Is this actively being worked on? In addition to the above (I take @thoas's word for it), the migrations need to be converted to support both Django 1.7 (using Django migrations) and <= Django 1.6 using south_migrations.

Happy to create a patch - but this PR looks out of date.

**Comment by thoas** [comment]
@ashchristopher this PR is up to date and synced with the master of @jsocol

**Comment by ashchristopher** [comment]
@thoas yes - but you are missing move of the `migrations` folder to `south_migrations` and then generation of new `migrations`.

**Comment by thoas** [comment]
@ashchristopher you are right, will do it.

**Comment by thoas** [comment]
Migrations have been moved to `south_migrations` with a mention in docs.

I'm still getting test errors from travis, when I execute tox in local all environments passed.

screen shot 2014-10-17 at 10 38 31 am

**Comment by thoas** [comment]
@jsocol could we have your input on this PR plz?

**Comment by jsocol** [comment]
Hey all, sorry I haven't had a chance to dig into this. I'm going to carve out a day next weekend for catching up on PRs. Then I can dive in and look at getting this and a few other things merged.

File: docs/installation.rst [hunk 2]
```diff
@@ -29,7 +29,14 @@ want it *below* any middleware that tweaks cookies before sending them
 out.
 
 If you're using South_, you can run ``manage.py migrate`` to create the
-necessary tables. If not, you'll need to create and run the schema
+necessary tables, you'll need to customize the ``SOUTH_MIGRATION_MODULES``
+setting: ::
+
+    SOUTH_MIGRATION_MODULES = {
```
Comment (jsocol) [inline docs/installation.rst:35]: We need to add this setting to test_settings, too.

File: waffle/models.py [hunk 3]
```diff
@@ -22,8 +28,8 @@ class Flag(models.Model):
         'other settings. Leave as Unknown to use normally.'))
     percent = models.DecimalField(max_digits=3, decimal_places=1, null=True,
                                   blank=True, help_text=(
-        'A number between 0.0 and 99.9 to indicate a percentage of users for '
-        'whom this flag will be active.'))
+                                      'A number between 0.0 and 99.9 to indicate a percentage of users for '
```
Comment (jsocol) [inline waffle/models.py]: Not thrilled about fixing one pep8 error by introducing another here.

**Comment by jsocol** [comment]
There is a lot going on here. I'm not sure which parts are necessary for 1.7:
- I'm guessing dropping django-nose/nose was for django-nose's lack of 1.7 support.
- Were the settings and `keyfmt` moves necessary?
- Were the signal handler moves necessary?
- There are a bunch of outstanding pep8 issues (line length mostly) introduced here. (I'm ignoring the management commands and migrations.)
- We need 1.7-compatibile initial migrations. It'd be best to add it to to `run.sh schema`

**Comment by thoas** [comment]
- Yes, the drop of django-nose/nose was due to a lack of 1.7 support and this runner was definitely not needed
- I have moved `keyfmt` to its own file because it's an utility method, I can revert if you think it's not necessary
- The move of signals was necessary, importing models logic in `__init__.py` can fails when using custom user (apps not ready, etc.)
- I have fixed the pep8 issues
- I have added `makemigrations` task to run.sh

**Comment by thoas** [comment]
@jsocol I have synced my branch with the current master.

**Comment by jsocol** [comment]
Do you have any idea what's up with the test failures? They seem like actual failures, not errors, but they are crazy sporadic over the past few builds.

**Comment by thoas** [comment]
All tests passed on tox in all environments in local, I do not have any idea of what's going wrong with travis.

I have spend a lot of time but haven't found anything. You are right on the fact there are actually failures.

**Comment by thoas** [comment]
Any follow-up on this PR?

@jsocol could you tell me if you are planning to merge it or not (no big deal if not ;))?

Thank you.

**Comment by jsocol** [comment]
I can't merge it with the inconsistent test failures.

This got pretty big and gnarly. I'd rather take the work done here and spread it out over smaller commits/pull reqs. E.g. drop django-nose without touching anything else, move the signal handlers, do the setting refactor, etc, all separately, before making the actual 1.7 changes last.

Hopefully we just won't run into the test issues, but if we do, it should be a lot easier to suss out what's wrong in smaller branches.

I actually have a little time this week so I'm happy to do that, taking this pull req as a guide. So I'm going to close out this pull req and I might cherry-pick commits out of it, but even if I can't, thank you for doing all the leg work of figuring out what related changes need to happen!

**Comment by thoas** [comment]
Ok I understand, no problem.

Let me get some time this week to dispatch this PR into multiple PRs as you said.

I will ping you.

**Pull request closed without merging**